lib/RefBackend/LowerToLLVM.cpp

```diff
@@ -38,21 +38,59 @@ static LLVMPointerType getInt8PointerType(MLIRContext *context) {
   return LLVMPointerType::get(IntegerType::get(context, 8));
 }
 
+static LLVMPointerType getInt64PointerType(MLIRContext *context) {
+  return LLVMPointerType::get(IntegerType::get(context, 64));
+}
+
+static LLVMStructType getInputDescriptorTy(MLIRContext *context) {
+  return LLVMStructType::getLiteral(context, {
+    // ArgType
+    IntegerType::get(context, 32),
+    // ElementType
+    IntegerType::get(context, 32),
+    // Rank
+    IntegerType::get(context, 32),
+    // Extents
+    LLVMPointerType::get(IntegerType::get(context, 64)),
```

Comment:
we have been using i32 for extents. (certain large models require larger, but for now this seems fine, and lets us scale down more easily to e.g. microcontrollers (it eventually needs to be a "target flag" of some kind))